packages/plugin-page-view-tracking-browser/src/page-view-tracking.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  BaseEvent,
+  BrowserClient,
+  BrowserConfig,
+  EnrichmentPlugin,
+  Event,
+  IdentifyOperation,
+  IdentifyUserProperties,
+  PluginType,
+} from '@amplitude/analytics-types';
+import { BASE_CAMPAIGN } from './constant';
+import { PageTrackingBrowserOptions } from './typings/page-view-tracking';
+
+export const pageViewTrackingPlugin = (
+  instance: BrowserClient,
+  options: PageTrackingBrowserOptions = {},
+): EnrichmentPlugin => {
+  let pageTrackingConfig = options;
+  return {
+    name: 'page-view-tracking',
+    type: PluginType.ENRICHMENT,
+
+    setup: async (config: BrowserConfig) => {
+      const attributionConfig = config.attribution;
+      pageTrackingConfig = {
+        trackOn: attributionConfig?.trackPageViews ? 'attribution' : undefined,
+        ...pageTrackingConfig,
+      };
+
+      // Turn off sending page view event by "runAttributionStrategy" function
+      if (attributionConfig) {
+        attributionConfig.trackPageViews = false;
+      }
+
+      if (
+        typeof pageTrackingConfig.trackOn === 'undefined' ||
+        (typeof pageTrackingConfig.trackOn === 'function' && pageTrackingConfig.trackOn())
+      ) {
+        instance.track(createPageViewEvent());
+      }
+    },
+
+    execute: async (event: Event) => {
+      if (pageTrackingConfig.trackOn === 'attribution' && isCampaignEvent(event)) {
+        const pageViewEvent = createPageViewEvent();
+        event.event_type = pageViewEvent.event_type;
```

Comment:
Hi @dnshi, any chance that you still remember why we override the event type from identify to page view? It feel makes more sense to keep the identify event and append attributions to the page view's event properties so that the identify event can actually set the user properties.

Reply:
Basically if the event is a campaign event, then the event type should match page view event type. If the event is not a campaign event or SDK does not track on attribution, then this condition is skipped.

Reply:
> Basically if the event is a campaign event, then the event type should match page view event type.
In this case, there will be no identify event and the user properties will not update the user properties map in the backend and not take effect in the following events. Note that only `$identify` event can set user properties.

Reply:
I think there were some changes, which were different from the version I worked on. Feel free to update the code to match current scenarios. Thanks.